opengrok-tools/src/main/python/opengrok_tools/utils/mirror.py

```diff
@@ -0,0 +1,400 @@
+                                          project_config.
+                                          get(CMD_TIMEOUT_PROPERTY))
+        if project_command_timeout:
+            command_timeout = project_command_timeout
+            logger.debug("Project command timeout = {}".
+                         format(command_timeout))
+
+        project_hook_timeout = get_int(logger, "hook timeout for "
+                                               "project {}".
+                                       format(project_name),
+                                       project_config.
+                                       get(HOOK_TIMEOUT_PROPERTY))
+        if project_hook_timeout:
+            hook_timeout = project_hook_timeout
+            logger.debug("Project hook timeout = {}".
+                         format(hook_timeout))
+
+        ignored_repos = project_config.get(IGNORED_REPOS_PROPERTY)
+        if ignored_repos:
+            logger.debug("has ignored repositories: {}".
+                         format(ignored_repos))
+
+        hooks = project_config.get(HOOKS_PROPERTY)
+        if hooks:
+            for hookname in hooks:
+                if hookname == "pre":
+                    prehook = os.path.join(hookdir, hooks['pre'])
+                    logger.debug("pre-hook = {}".format(prehook))
+                elif hookname == "post":
+                    posthook = os.path.join(hookdir, hooks['post'])
+                    logger.debug("post-hook = {}".format(posthook))
+
+        if project_config.get(PROXY_PROPERTY):
+            logger.debug("will use proxy")
+            use_proxy = True
+
+    if not ignored_repos:
+        ignored_repos = []
+
+    return prehook, posthook, hook_timeout, command_timeout, \
+        use_proxy, ignored_repos
+
+
+def mirror_project(config, project_name, check_incoming, uri,
+                   source_root):
+    """
+    Mirror the repositories of single project.
+    :param config global configuration dictionary
+    :param project_name: name of the project
+    :param check_incoming:
+    :param uri
+    :param source_root
+    :return exit code
+    """
+
+    logger = logging.getLogger(__name__)
+
+    project_config = get_project_config(config, project_name)
+    prehook, posthook, hook_timeout, command_timeout, use_proxy, \
+        ignored_repos = get_project_properties(config, project_config,
+                                               config.get(HOOKDIR_PROPERTY))
+
+    proxy = None
+    if use_proxy:
+        proxy = config.get(PROXY_PROPERTY)
+
+    # We want this to be logged to the log file (if any).
+    if project_config:
+        if project_config.get(DISABLED_PROPERTY):
+            logger.info("Project {} disabled, exiting".
+                        format(project_name))
+            return CONTINUE_EXITVAL
+
+    #
+    # Cache the repositories first. This way it will be known that
+    # something is not right, avoiding any needless pre-hook run.
+    #
+    ret = 0
```

Comment:
you start to use symbolic values here, can you convert 0 and 1 to some names as well?